validator/anticorruption/src/main/java/tech/pegasys/artemis/validator/anticorruption/BeaconChainEventAdapter.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.validator.anticorruption;
+
+import com.google.common.eventbus.Subscribe;
+import com.google.common.primitives.UnsignedLong;
+import tech.pegasys.artemis.statetransition.events.attestation.BroadcastAttestationEvent;
+import tech.pegasys.artemis.util.time.channels.SlotEventsChannel;
+import tech.pegasys.artemis.validator.api.ValidatorTimingChannel;
+
+/**
+ * Converts events from the {@link com.google.common.eventbus.EventBus} to the new validator client
+ * {@link tech.pegasys.artemis.events.EventChannels}.
+ */
+class BeaconChainEventAdapter implements SlotEventsChannel {
+
+  private final ValidatorTimingChannel validatorTimingChannel;
+
+  public BeaconChainEventAdapter(final ValidatorTimingChannel validatorTimingChannel) {
+    this.validatorTimingChannel = validatorTimingChannel;
+  }
+
+  @Subscribe
+  public void onBroadcastAttestationEvent(final BroadcastAttestationEvent event) {
+    validatorTimingChannel.onAttestationCreationDue(event.getNodeSlot());
+  }
+
+  @Override
+  public void onSlot(final UnsignedLong slot) {
+    validatorTimingChannel.onSlot(slot);
```

Comment:
The block production duty is probably more time-dependent than the duty scheduling that gets performed in `onSlot` so I would reverse their order (unless there is a scenario in which we schedule the block production of the current slot in the exact slot itself, although since we're scheduling an epoch ahead, that probably should not happen)
```suggestion
    validatorTimingChannel.onBlockProductionDue(slot);
```

Reply:
Or, can we run them in parallel?

Reply:
The requests for both very quickly become asynchronous anyway because of the calls into `ValidatorApiChannel`.  Which is somewhat unfortunate because you need to have found the duties to perform before you know you need to produce a block which means we'll miss creating a block on the first slot we see because the duties won't yet be scheduled.  I think I can live with that though - if we'd started slightly later we'd have missed the slot anyway.